lib/github-provider.js

```diff
@@ -120,8 +120,104 @@ function mapGithubReleaseDate(camp, githubApiUrl, githubAuth) {
     }));
 }
 
+function mapGithubPRState(camp, githubApiUrl, githubAuth){
+  camp.route(/^\/github\/pr-state\/([^/]+)\/([^/]+)\/([^/]+)\.(svg|png|gif|jpg|json)$/,
+  cache((data, match, sendBadge, request) => {
+    const [, owner, repo, number, format] = match;
+    const apiUrl = `${githubApiUrl}/repos/${owner}/${repo}/pulls/${number}`;
+    const badgeData = getBadgeData('pull request', data);
+
+    if (badgeData.template === 'social') {
+      badgeData.logo = getLogo('github', data);
+    }
+
+    githubAuth.request(request, apiUrl, {}, function (err, res, buffer) {
+      if (err != null) {
+        badgeData.text[1] = 'inaccessible';
+        sendBadge(format, badgeData);
+        return;
+      }
+
+      //github return 404 if repo not found or no pr
+      if(res.statusCode === 404) {
+        badgeData.text[1] = 'no pr or repo not found';
+        sendBadge(format, badgeData);
+        return;
+      }
+
+      try {
+        const pr = JSON.parse(buffer);
+        let colorscheme = '';
+        let status = '';
```

Comment:
It would be clearer what's happening with these declarations `let colorscheme` if you moved them just before their switch statements. Also you can just write `let colorscheme;`. It doesn't help to assign an empty string.